### DISCUSSION TITLE
Feedback on Copilot after using it for more than two weeks

### DISCUSSION THREAD

**Discussion by RafidMuhymin** [community/community#8264] [Copilot Conversations]
It's been more than two working weeks that I'm using **GitHub Copilot**. I have to say it's the best AI code autocompletion tool I have used. Instead of talking in paragraphs, I'm going to list my thoughts on it.

The features that make *Copilot* the best:
* The code autocompletion is really the best. It doesn't need any further discussion to prove it.
* It's much faster than the other tools I have used. It doesn't matter whether you have a 10year old laptop or a modern gaming rig!
* It's not only good for programmers but content creators also. I really found great help when I was writing `markdown` documentation, blogs, and answering support questions.
* It's not restricted to any file extensions.
* It suggests good autocompletion for the new file extensions by using the context and mixing with other languages. The `Astro` component syntax is a good example where *Copilot* performs really well.

The features I expect from *Copilot* in the future:
* It should improve the suggestion when rewriting the code. I think the only way to do is by suggesting code completion based on the complete file. Or if not possible then it should at least read the next few lines (or finish the current code block).
* It should also provide completion inside general strings
* It should read the `clipboard` when providing suggestions

**Comment by ryanjsalva** [comment]
Thanks for the excellent feedback, @RafidMuhymin! Can you expand a little on the first two bullet points of what you expect in the future:

> It should improve the suggestion when rewriting the code. I think the only way to do is by suggesting code completion based on the complete file. Or if not possible then it should at least read the next few lines (or finish the current code block).

I _think_ you're saying that Copilot currently provides less-than-helpful suggestions when you're editing in the middle of a file, but maybe you're implying something different? Can you say more?

> It should also provide completion inside general strings

I _think_ you're saying that Copilot should provide suggestions when your cursor is in the middle of a token (e.g. a word, variable name, method name). Again, am I correctly understanding your intent here?

**Comment by RafidMuhymin** [comment]
@ryanjsalva Yeah, you are right on the first point. Copilot provides suggestions based on the previous lines of code. But it doesn't take the next lines into account. So, when I am rewriting a file most of the suggestions shown are incorrect.

And I see Copilot already provides suggestions inside general strings. When I suggested this feature I was getting suggestions only inside `` backticks (I primarily write JS).

And I'll wait for your opinion on my third point.
